### PR TITLE
Fix setState warning#2061,remove jQuery#2065 in StaticAppBar

### LINKED ISSUE #2061: Fix setState warning in StaticAppBar, TopBar component
**Actual Behaviour**

Due to setState in staticAppBar, topBar, componentDidMount tries to set state, even on unmounted bar, error occurs

**Expected Behaviour**

Should not be present

**LogCat for the issue**

> proxyConsole.js:56 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the StaticAppBar component.

**Screenshots of the issue**

screenshot 2019-03-03 at 2 53 06 pm

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #2061, #2065 

Changes: 

- Fix setState warning

screenshot 2019-03-03 at 2 53 06 pm

- Remove jQuery dependency for DOM manipulation

Demo Link: https://pr-2066-1-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

File: src/components/StaticAppBar/StaticAppBar.react.js [hunk 1]
```diff
@@ -148,34 +147,52 @@ class StaticAppBar extends Component {
...
       didScroll = true;
-      this.setState({ isPopUpMenuOpen: false });
+      if (this.mounted) {
+        this.setState({ showOptions: false });
+      }
     });
 
     const hasScrolled = () => {
-      let st = $(window).scrollTop();
+      let st = document.scrollingElement.scrollTop;
```
Comment (AakashMallik) [inline src/components/StaticAppBar/StaticAppBar.react.js]: A more descriptive variable name would be nice I guess. **st**

Comment (AakashMallik) [reply src/components/StaticAppBar/StaticAppBar.react.js]: @fragm3 any updates on this?

**Comment by fragm3** [comment]
@AakashMallik made the change, @akshatnitd please review

File: src/components/StaticAppBar/StaticAppBar.react.js [hunk 2]
```diff
@@ -148,36 +147,58 @@ class StaticAppBar extends Component {
     }
   };
 
+  componentWillUnmount() {
+    this.mounted = false;
```
Comment (akshatnitd) [inline src/components/StaticAppBar/StaticAppBar.react.js]: Also remove event listener on component unmount.

**Comment by fragm3** [comment]
@akshatnitd made the changes

**Pull request merged**